src/routes/transactions/transactions.service.ts

```diff
@@ -218,6 +221,57 @@ export class TransactionsService {
     };
   }
 
+  private getAdjustedPaginationForHistory(
+    paginationData: PaginationData,
+  ): PaginationData {
+    if (paginationData.offset > 0) {
+      return new PaginationData(
+        paginationData.limit + 1,
+        paginationData.offset - 1,
+      );
+    }
+    return paginationData;
+  }
+
+  async getTransactionHistory(
+    chainId: string,
+    routeUrl: Readonly<URL>,
+    safeAddress: string,
+    paginationData: PaginationData,
+    timezoneOffset?: string,
+  ): Promise<TransactionItemPage> {
+    const paginationDataAdjusted =
+      this.getAdjustedPaginationForHistory(paginationData);
+    const domainTransactions = await this.safeRepository.getTransactionHistory(
+      chainId,
+      safeAddress,
+      paginationDataAdjusted.limit,
+      paginationDataAdjusted.offset,
+    );
+    const nextURL = buildNextPageURL(routeUrl, domainTransactions.count);
+    const previousURL = buildPreviousPageURL(routeUrl);
+    if (nextURL == null) {
+      const creationTransaction =
+        await this.safeRepository.getCreationTransaction(chainId, safeAddress);
+      domainTransactions.results.push(creationTransaction);
+    }
+    const safeInfo = await this.safeRepository.getSafe(chainId, safeAddress);
+    const results = await this.transactionMapper.mapTransactions(
+      chainId,
+      domainTransactions.results,
+      safeInfo,
+      paginationData.offset,
+      parseInt(timezoneOffset ?? '0'),
```

Comment:
In this case, instead of doing this parsing manually, I'd suggest parsing this at the controller level. I think the domain layer should receive a `number`. I mean, at the controller level you can parse by just defining the type as `number` or using Nest built-in `ParseIntPipe` (with a default value if needed), it's really simple to add: https://docs.nestjs.com/pipes
```javascript
@Query('timezone_offset', new DefaultValuePipe(0), ParseIntPipe) timezoneOffset: number,
```

Reply:
Done